activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -167,6 +167,58 @@ def attribute_names
         end
       end
 
+      # An Array of String attribute names and aliases for accessing those
+      # attributes.
+      #
+      #   class Person < ActiveRecord::Base
+      #   end
+      #
+      #   Person.attribute_names_and_aliases
+      #   # => ["id", "created_at", "updated_at", "name", "age"]
+      def attribute_names_and_aliases
+        @attribute_names_and_aliases ||= attribute_names + attribute_aliases.keys
+      end
+
+      # Can the given name be treated as a column name? Returns true if name
+      # is attribute or attribute alias.
+      #
+      #   class Person < ActiveRecord::Base
+      #   end
+      #
+      #   Person.respond_to_attribute?(:name)
+      #   # => true
+      #
+      #   Person.respond_to_attribute?("foo")
+      #   # => false
+      def respond_to_attribute?(name)
```

Comment:
This method needs to be public to be called from `calculations.rb`. Did you just mean I should add `# :nodoc:`?